src/components/Timeslot/index.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Redirect } from "react-router-dom";
+
+import { postBookingThunk } from "../../store/bookings/actions";
+import { selectBookingSuccess } from "../../store/bookings/selectors";
+
+import { IoIosMore } from "react-icons/io";
+
+export default function Timeslot(props) {
+  // console.log("prop timeslot", props);
+  const [show, setShow] = useState(false);
+  const [extra, setExtra] = useState(0);
+  const [extraInputs, setExtraInputs] = useState([]);
+  const dispatch = useDispatch();
+  const book = useSelector(selectBookingSuccess);
+  // console.log("book", book);
+
+  if (book) {
+    return <Redirect to="/gyms/:id/booking/:bookingId" />;
+  }
+
+  const onExtraNameChange = (i, name) => {
+    const newExtraInputs = extraInputs.slice();
+    newExtraInputs[i] = name;
+    setExtraInputs(newExtraInputs);
+  };
+
+  const showDetails = () => {
+    let inputs = [];
+    const numberExtra = parseInt(extra);
+
+    if (numberExtra > 0) {
+      for (let i = 0; i < numberExtra; i++) {
+        inputs.push(
+          <input
+            type="text"
+            placeholder="Name partner"
+            key={[i]}
+            value={extraInputs[i]}
+            onChange={(e) => onExtraNameChange(i, e.target.value)}
+            className="extraClimbers"
+          />
+        );
+      }
+    }
+
+    function handleSubmit(e) {
+      e.preventDefault();
+
+      const finalPartners = extraInputs.slice(0, extra);
+      // console.log(finalPartners);
+
+      const namePartner = finalPartners;
+      //timeslotId is not the same as the actual id in the table!
+      const timeslotId = props.info.id;
+      const gymId = props.info.gymId;
+
+      // console.log("dispatch", namePartner, timeslotId, gymId, userId);
+      dispatch(postBookingThunk(namePartner, timeslotId, gymId));
+      setExtra(0);
+      setShow(false);
+      setExtraInputs([]);
+    }
+
+    return (
+      <div>
+        <form onSubmit={handleSubmit}>
+          <label>Extra climber(s): </label>
+          <input
+            type="number"
+            min="0"
+            max="3"
+            value={extra}
+            onChange={(e) => {
+              setExtra(e.target.value);
+            }}
+          />
+          {inputs}
+
+          <button type="submit">Book session</button>
+        </form>
+      </div>
+    );
+  };
+
+  function clickHandle() {
+    setShow(!show);
+  }
+
+  return (
+    <div key={props.info.id} className="timeslot">
+      <h4>
+        {parseFloat(props.info.startTime).toFixed(2)} -{" "}
+        {parseFloat(props.info.endTime).toFixed(2)}
+      </h4>
+      <p>Max Capacity: {props.info.maxCap}</p>
+      <p>Spots available: {props.info.maxCap - 2}</p>
+      {show && showDetails()}
+      <div onClick={clickHandle} key={props.info.id} className="extraBtn">
```

Comment:
clickHandle not really necessary you can do straight away:
```
<div onClick={() => setShow(!show)} key={props.info.id} className="extraBtn">
```
If it's simple, do it inline